keras/backend/jax/distribution_lib.py

```diff
@@ -40,7 +67,13 @@ def distribute_value(value, tensor_layout):
     """
     if not isinstance(tensor_layout, jax.sharding.Sharding):
         tensor_layout = _to_jax_layout(tensor_layout)
-    return jax.device_put(value, tensor_layout)
+
+    # TODO(scottzhu): This might not be a cheap check, we should consider
+    # have some proper JAX API for doing this check.
+    if jax_utils.is_in_jax_tracing_scope():
+        return jax.lax.with_sharding_constraint(tensor, tensor_layout)
+    else:
```

Comment:
You can remove the indent block and just do `return`

Reply:
Done.